### ISSUE TITLE
Interested to deploy on a L2-Blockchain?

### ISSUE THREAD

**Issue by wsdt** [tytzM17/Lokian.eth#5]
Hi, 
one of my customers is running a L2 chains and it seems like you are actively implementing this game for a while now. 

Would you be interested to launch this game on their chain? We could jump onto a call in the next few days if you like. 

Thanks,
Kevin, Wavect GmbH

**Comment by tytzM17** [comment]
hello, 

yeah Sure , why not, I'm interested ;)  Is it evm compatible like Polygon or Arbitrum  ? currently my dapp is on the polygon testnet mumbai.

Thank u for reaching out.
-tytzM@Lokian,eth

**Comment by wsdt** [comment]
Hi, 
ok cool! Yes it is EVM compatible. Their solution even supports api calls directly from a smart contract without the need of using a callback like with ChainLink. 

If you have time, we could do a quick call right now if you want? 

My e-mail: kevin.riedl@wavect.io

**Comment by wsdt** [comment]
Hi my friend, 
just trying to play around with it. 

It says sprites not found as you are requesting the folder './sprites-copy/'. I just copied the sprites folder and named it like that but the naming is also different (having in sprites just a numerical naming scheme and in sprites-copy expecting a descriptive name). 

Is there anything I'm missing? Looking forward to your anser :)

**Comment by wsdt** [comment]
Yes you have it in the gitignore `/src/sprites-copy/*`. 

Because of copyright, bc you used pokemon graphics? 

Would it be possible to send the graphics to me via e-mail just for testing? We will replace them with some of our NFTs if that's okay for you.

**Comment by wsdt** [comment]
but as said, just playing around for now, need to evaluate if that's what we are looking for.

**Comment by tytzM17** [comment]
hi ther, it's here now --> https://tytzm17.github.io/Lokian.eth/

**Comment by tytzM17** [comment]
give me ur mumbai testnet public address and i'll send u some erc20 tokens to buy some NFTs ;)

**Comment by wsdt** [comment]
ah cool thx! But any chance that I get the actual folder from you, so that I can try to run it locally as well? :)

**Comment by tytzM17** [comment]
yes coz of copyright, i have changed the creature graphics, i think you can run the repo, buy renaming the folder 'sprites' to 'sprites-copy' ?

**Comment by wsdt** [comment]
ok no problem, yes tried that but your code searches for different names (the sprites folder has only images like 0.png, etc.)

**Comment by tytzM17** [comment]
okay let me check and i'll get back at u. thanks for the interest in my project. what L2 customers do u have, is it like polygon or arbitrum etc?

**Comment by tytzM17** [comment]
i have made some changes, i think it would use the pokemn graphics now ;)

**Comment by wsdt** [comment]
it's Boba, a hybrid computing L2. Quite new to the market :)
I'll play around a little and keep you updated. If desired I would work with you to make the project maybe a success on their network?

**Comment by wsdt** [comment]
thanks! Will try later

**Comment by tytzM17** [comment]
hmmi've heard of them. seems like a new L2 ;) .. alright ,thanks, u can now just npm i and npm start. just msg me for ur feedback thanks again :)

**Comment by wsdt** [comment]
ok great thanks! just tried it out on their L2 and it runs smoothly! 
They actually have a very simple NFT collection (which the community was able to mint for free). 

Would you be down to replace the pokemon images with their so-called "TuringMonsters" NFTs? So you would need to own these NFTs to trade them etc (basically this might work out of the box right, by simply replacing the nft contract + the images in the sprites folder?). 

Would support you here and help you to deploy it on their network (for sure with full support from our side). Your game could have some traction in their community :)

**Comment by tytzM17** [comment]
hello, I think that could possibly work too, but needs modification on my part coz currently the monsters/creatures are not of nft type ;)

**Comment by tytzM17** [comment]
ur pr was merged ;) let me also try to implement the 'TuringMonsters' into the game. is it erc1155? hopefully it would work.

**Comment by wsdt** [comment]
It's a regular erc721. 

Here is the mainnet deployment: 
https://blockexplorer.boba.network/address/0xce458FC7cfC322cDd65eC77Cf7B6410002E2D793/transactions

Open Source code: 
https://github.com/bobanetwork/boba/tree/develop/boba_community/turing-monsters

Let me know if I can help :)

**Comment by tytzM17** [comment]
alright ,I 'll check it out and try to implement the nfts to the game. i'll also let u know if I need some help. thanks for this, much appreciated ;)

**Comment by wsdt** [comment]
Ok great mate! Looking forward to it!

**Comment by wsdt** [comment]
Hi mate, how is it going?

**Comment by tytzM17** [comment]
hello there, I have not yet finished the implementation of the turing monsters but I am getting there.

**Comment by wsdt** [comment]
ok cool! Yeah anything I can help you with? We also would like to distribute your game then to our community etc.

**Comment by tytzM17** [comment]
yes, that would be nice. I would be interested. What would u suggest I would do to get the nfts of the users who connect with wallet like metamask. I was thinking of using Moralis for rinkeby to get the user's nft, but they dont have boba or l2 yet for mainnet. I would then use the tumo stats for the monster duels, and also other fun features in the future.

**Comment by wsdt** [comment]
Hi yes unfortunately Moralis is not yet supported, but we are on it. 

Nevertheless, checking of ownership etc is quite easy achievable without Moralis or what kind of data would you need? Maybe I have some good ways around that.

**Comment by tytzM17** [comment]
i would like to know the attributes of the different nft/ tumo, and also the nft the user/address got , just like in etherscan or polygon scan which  knows what tokens and nfts you have. thank u

**Comment by tytzM17** [comment]
hi , i think the api of boba's blockscout could get the nfts.. this one 'Get list of tokens owned by address.
?module=account&action=tokenlist&address={addressHash}' at https://blockexplorer.boba.network/api-docs

**Comment by wsdt** [comment]
Yeah sure mate, here are some details: 

Boba-Rinkeby (Testnet) 
* Explorer: https://blockexplorer.rinkeby.boba.network/
* For TuringMonsters NFT just use a regular ERC721

Mainnet 
* Explorer: https://blockexplorer.boba.network/

* Turing Monsters-NFT: https://blockexplorer.boba.network/address/0xce458FC7cfC322cDd65eC77Cf7B6410002E2D793/transactions

* Turing Monsters Collection auf NFT Marketplace: https://shibuinft.com/app/collection/0xce458fc7cfc322cdd65ec77cf7b6410002e2d793

I hope this helps mate! 

Further documentation: 
* Other contract addresses for mainnet & testnet: https://github.com/bobanetwork/boba/tree/develop/packages/boba/register/addresses

* General documentation: https://boba.network/start-here/

* Gateway to bridge testnet tokens from Rinkeby to Boba Rinkeby: https://gateway.rinkeby.boba.network/

**Comment by wsdt** [comment]
Good catch yes could work. 

We also have a subgraph on TuringMonsters (also for testnet): 
https://github.com/bobanetwork/boba/tree/develop/packages/boba/subgraph

Rinkeby GraphQL endpoint: https://graph.rinkeby.boba.network/
Mainnet endpoint: https://thegraph.com/hosted-service/subgraph/bobanetwork/boba-l2-subgraph

**Comment by tytzM17** [comment]
> Yeah sure mate, here are some details:
> 
> Boba-Rinkeby (Testnet)
> 
> * Explorer: https://blockexplorer.rinkeby.boba.network/
> * For TuringMonsters NFT just use a regular ERC721
> 
> Mainnet
> 
> * Explorer: https://blockexplorer.boba.network/
> * Turing Monsters-NFT: https://blockexplorer.boba.network/address/0xce458FC7cfC322cDd65eC77Cf7B6410002E2D793/transactions
> * Turing Monsters Collection auf NFT Marketplace: https://shibuinft.com/app/collection/0xce458fc7cfc322cdd65ec77cf7b6410002e2d793
> 
> I hope this helps mate!
> 
> Further documentation:
> 
> * Other contract addresses for mainnet & testnet: https://github.com/bobanetwork/boba/tree/develop/packages/boba/register/addresses
> * General documentation: https://boba.network/start-here/
> * Gateway to bridge testnet tokens from Rinkeby to Boba Rinkeby: https://gateway.rinkeby.boba.network/

yes, I have tried the gateway and bought some eth and boba tokens too, to play around. thanks for this. I appreciate it.

**Comment by wsdt** [comment]
@tytzM17 Hi mate, talked with the Boba team. Would you be up for a call with them to see what you need to do that project? 
Do you have a Telegram handle or E-mail for us?

**Comment by tytzM17** [comment]
yes, I have . for telegram its https://t.me/m1dNytBluu and email is official@lokian.monster or ub17@protonmail.com. thanks

**Comment by wsdt** [comment]
thanks! Forwarded your contact details. You should receive a mail or Telegram soon.

**Comment by tytzM17** [comment]
alright, thank you very much 😸
